### ISSUE TITLE
Missing weight for eigen_centrality()

### ISSUE THREAD

**Issue by cccfran** [igraph/rigraph#375]
How does eigen_centrality() handle missing weights when using the weight option? Does it treat NA as 0? Thanks.

**Comment by vtraag** [comment]
The issue tracker is not meant for support questions but for bug reports. The support forum can be found at https://igraph.discourse.group. You can probably better post your question there.

**Comment by vtraag** [comment]
Following our discussion [here](https://igraph.discourse.group/t/r-missing-weight-for-eigen-centrality/144) this issue does seem to be relevant. In particular, `NA` is not treated correctly at the moment, and should be corrected. In particular
```R
  if (!is.null(weights) && any(!is.na(weights))) { 
  weights <- as.numeric(weights) 
  } else { 
  weights <- NULL 
  }
```
is incorrected, and the condition should instead be all(!is.na(weights)) or !any(is.na(weights)). This should be corrected in the `types-R.def` file.

**Comment by ntamas** [comment]
My two cents (also posted in the discussion forum): I think that the current treatment of `NA` values is correct. `NA` is commonly used to represent missing data, and we should not ignore a weight vector silently just because it happens to contain an `NA` somewhere in the middle. It should be the duty of the C layer to handle NaNs correctly; R should simply pass whatever it has been given to the C layer.

Right now what probably happens is that `NA` gets converted to `NaN`, which then goes to the C layer, which passes it on to ARPACK, which then probably propagates `NaN` all over the place in the result. This is somewhat undesired, but it should be the job of the C layer to bail out early when it sees a `NaN` in the input weight vector. (For other algorithms, a lingering `NaN` might not be a problem; for instance, one can happily calculate weighted shortest paths in a connected component of a graph if some _other_ component contains a `NaN` weight).

**Comment by ntamas** [comment]
Anyhow, closing this issue as it was more like a support question and it has been answered above.

**Comment by vtraag** [comment]
Yes, my previous reply was outdated indeed. Many of these elements are now addressed in the C core by improved NA handling, but I think that's included only from some 0.9 version onwards.
